core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1733,15 +1733,17 @@ public static function addImageToTemplate($objTemplate, $arrItem, $intMaxWidth=n
 		// Image link
 		if (TL_MODE == 'FE' && $arrItem['imageUrl'])
 		{
-			$objTemplate->$strHrefKey = $arrItem['imageUrl'];
+			$imageUrl = urldecode(self::replaceInsertTags($arrItem['imageUrl']));
```

Comment:
I don’t think we can use `urldecode` here because it would potentially break external URLs.

Reply:
Hm, yeah, the problem is that we generally cannot know, whether the content in `$arrItem` is url encoded or not, no matter if it comes from an Insert Tag or if the URL is put in manually.

Reply:
I have updated the PR to only specially handle `{{file::*}}` insert tags instead, where we know the output is url encoded and can be safely decoded.

Reply:
I don't like this solution either though, because it will only handle `{{file::*}}` properly. But if you set the **Image link target** to `files/foo%20bar/foo%20bar.jpg` for example, then this file is again not processed by Contao, even though it could.

Should we instead simply url decode and encode all **Image link target**s that are _not_ external?

Reply:
I think we should use the same logic as in #2716. First check if the URL is external, if not, urldecode and check if the path exists on disk.

Reply:
Should we always replace all insert tags though? I don't know how this is handled in the Image Studio.

Reply:
Replacing insert tags is expensive and should only be done if really necessary.

Reply:
> Replacing insert tags is expensive and should only be done if really necessary.

But we already do it in Contao 4.10 and up: https://github.com/contao/contao/pull/2674#issuecomment-766356141

Reply:
I see. We should not do it there, either (if possible).

Reply:
We could limit it to `{{file::*}}`. But then any custom insert tag that would return a valid, url encoded path to an image will not get processed.

Reply:
IMO replacing all insert tags is required because of the possibility of custom insert tags (e.g. created via a picker).

> Replacing insert tags is expensive

Why is that? If `preg_split` dosn’t match anything is sould return immediately and should cost much performance IMO.

Reply:
> I don't know how this is handled in the Image Studio.

There is no replacing of insert tags in the Image Studio code but in the `ModelMetadataTrait`. There we *do* call `replaceInsertTags()` on each `Metadata::VALUE_URL` though in order to guarantee that we're dealing with valid paths.

Reply:
Ok, in this case I guess it is OK to replace all insert tags.